### DISCUSSION TITLE
Function signature and return type in the completion table

### DISCUSSION THREAD

**Discussion by Jlll1** [helix-editor/helix#10350] [Ideas]
One feature I really feel like I'm missing is ability to preview information about functions without selecting them to trigger the popup. Information such as return types / full function signatures are useful to filter out the ones you don't care about when exploring a codebase.

At the same time it does add a lot of clutter, so I think having this togglable would be great.

Some quick screenshots
**Full function signature**
![image](https://user-images.githubusercontent.com/71319302/147874772-9947682d-d350-4829-bb21-63e475b1371f.png)

**Only the return type**
![image](https://user-images.githubusercontent.com/71319302/147875270-14091ca8-2199-4d9a-ab22-3f7af8e013b0.png)

**Use displaying return type as indication it's a function**
![image](https://user-images.githubusercontent.com/71319302/147875329-6a560510-fd9e-42aa-895f-5ed3cd3f37d1.png)

**Comment by Jlll1** [comment]
Potentially it could be also expanded for things like fields, where instead of just having 'field' there, we'd have their type.

**Comment by matoous** [comment]
Or the signature help could be displayd on window as you type the parameters: https://github.com/helix-editor/helix/discussions/1329, or both 🤷 or both and make it configurable.

**Comment by Jlll1** [comment]
I think the signature help is another feature altogether. 
I treat this more as a preview of the api I have access to. If not the whole signature, having the return type for functions and type for fields is very useful to have and makes it much easier to filter out things you're not interested in in the autocomplete.

I wonder if maybe allowing for free configuration of the completion pop-up would be the way to go, or maybe just having couple layouts the user can choose from?
